sw/device/lib/base/freestanding/stddef.h

```diff
@@ -0,0 +1,26 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_LIB_BASE_FREESTANDING_STDDEF_H_
+#define SW_DEVICE_LIB_BASE_FREESTANDING_STDDEF_H_
+
+#include <stdint.h>
+
+/**
+ * This header implements the stdint.h standard header, as required by C11 S4p6.
+ * This header is specified in detail in S7.19 of the same.
+ *
+ * See https://github.com/riscv/riscv-elf-psabi-doc/blob/master/riscv-elf.md
+ * for the RISC-V specifics this file conforms to.
+ */
+
+typedef __INTPTR_TYPE__ ptrdiff_t;
+typedef __UINTPTR_TYPE__ size_t;
+typedef uint64_t max_align_t;
```

Comment:
clang/lib/Headers/__stddef_max_align_t.h shows a definition matching GCC. `long double` has larger alignment than `uint64_t` in the ILP32 psABI.

I note that LLVM's libcss stddef.h definition defaults to `typedef long double max_align_t` if the compiler didn't define one, which is probably a good example to follow.

Reply:
I went with a different strategy that uses the existing __BIGGEST_ALIGNMENT__ macro, but it's basically the same thing.

Reply:
Good idea, I like that solution.